<launch>
  <!-- Output color format is BGR8 -->
  
  <arg name="image_in" />
  <arg name="image_out" />
  <arg name="info_in" default="" />
  <arg name="info_out" default="" />
  <arg name="scale" default="0.5" />
  <arg name="debayer" default="false" />
  
  <arg name="publish_static_camera_info" default="false" />
  <!-- The following arguments are only necessary if publish_static_camera_info is true -->
  <arg name="camera_name" default="camera" />
  <arg name="camera_info_url" default="" />
  <arg name="scale_camera_info" default="false" />


  <arg name="pkg" value="image_transforms" />
  <arg name="name" default="image_scaler_$(arg camera_name)" />
  <arg name="type" default="image_scaler_node" />
  <arg name="rate" default="60" />
  <arg name="args" value="_loop_rate:=$(arg rate) _scale:=$(arg scale) _debayer:=$(arg debayer) _scale_camera_info:=$(arg scale_camera_info) _publish_static_camera_info:=$(arg publish_static_camera_info) _camera_name:=$(arg camera_name) _camera_info_url:=$(arg camera_info_url) ~image:=$(arg image_in) ~output_image:=$(arg image_out) ~camera_info_in:=$(arg info_in) ~camera_info_out:=$(arg info_out)" />
  <arg name="manager" default="manager" />
  <arg name="nodelet" default="false" />
  <node
      if="$(arg nodelet)"
      pkg="nodelet"
      type="nodelet"
      name="$(arg name)"
      args="load $(arg pkg)/$(arg name) $(arg manager) $(arg args)"
      output="screen" />
  <node
      unless="$(arg nodelet)"
      pkg="$(arg pkg)"
      type="$(arg type)"
      name="$(arg name)"
      args="$(arg args)"
      output="screen" />

</launch>
